packages/builder/src/index.ts

```diff
@@ -224,18 +224,16 @@ async function _lint(
       continue;
     }
 
-    const contents = getFileContents(file);
-
     /**
      * TODO: this was copied from TSLint builder - is it necessary here?
      */
     // Give some breathing space to other promises that might be waiting.
     await Promise.resolve();
-    lintReports.push(cli.executeOnText(contents, file));
     lintedFiles.add(file);
+    filesToLint.push(file);
   }
 
-  return lintReports;
+  return [cli.executeOnFiles(filesToLint)];
```

Comment:
only `executeOnFiles` persists the cache to disk: https://github.com/eslint/eslint/blob/master/lib/cli-engine/cli-engine.js#L821